In [2]:
import pandas as py
Monthly_Plan=py.read_csv(r"C:\Users\ADMIN\Desktop\Retail_Analytics_Project\02_Raw_Data\Monthly_Seasonal_Planning.csv")
print(Monthly_Plan.head().to_string())

      Month Site ID Product Category  Forecasted Sales  Actual Sales  Seasonal Adjustments
0  May-2024    PS3Y            Dairy          81260.50  96315.510888                0.1853
1  Dec-2024    W1Q5          Apparel          60609.93  64357.950697                0.0618
2  Sep-2024    T4UF          Apparel          84277.99  94950.573265                0.1266
3  Dec-2024    T4UF          Apparel          79067.48  82228.634271                0.0400
4  Mar-2025    OTOH          Apparel          50727.92  42559.446989               -0.1610


In [4]:
print("Monthly_Plan SHAPE")
print(Monthly_Plan.shape)

print("Monthly_Plan info")
print(Monthly_Plan.info())

print("Monthly_Plan describe")
print(Monthly_Plan.describe())



Monthly_Plan SHAPE
(120, 6)
Monthly_Plan info
<class 'pandas.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Month                 120 non-null    str    
 1   Site ID               120 non-null    str    
 2   Product Category      120 non-null    str    
 3   Forecasted Sales      120 non-null    float64
 4   Actual Sales          120 non-null    float64
 5   Seasonal Adjustments  120 non-null    float64
dtypes: float64(3), str(3)
memory usage: 5.8 KB
None
Monthly_Plan describe
       Forecasted Sales   Actual Sales  Seasonal Adjustments
count        120.000000     120.000000            120.000000
mean       56104.728000   55617.085492             -0.007223
std        23907.643229   24577.008349              0.111414
min        10647.460000    9809.208469             -0.199500
25%        33070.150000   35859.745391             -0.100500
50%        5

In [10]:
print("Monthly_Plan validation")

print("Unique Site IDs      :", Monthly_Plan["Site ID"].nunique())
print("Duplicate Site IDs   :", Monthly_Plan["Site ID"].duplicated().sum())
print("Null Site IDs        :", Monthly_Plan["Site ID"].isnull().sum())

Monthly_Plan validation
Unique Site IDs      : 47
Duplicate Site IDs   : 73
Null Site IDs        : 0


In [11]:
print("Duplicate Rows:", Monthly_Plan.duplicated().sum())

Duplicate Rows: 0


In [12]:
Monthly_Plan["Site ID"].value_counts()[
    Monthly_Plan["Site ID"].value_counts() > 1
]

Site ID
SV87    6
M5IG    5
KJLT    5
CX19    5
SEMC    5
AM14    5
ID25    4
MYZY    4
I3QK    4
T4UF    3
7EEQ    3
4CIY    3
IUC0    3
M5K8    3
NY20    3
5JSG    3
XDP5    3
H75L    3
A11D    3
PKH1    3
BVM2    3
WM31    3
CXUL    3
PS3Y    2
93TY    2
0T0P    2
2VLP    2
0RJI    2
FFYV    2
ZYTJ    2
A1WJ    2
EXQ7    2
30T9    2
4ZKL    2
Name: count, dtype: int64

In [15]:
Monthly_Plan["Month"] = py.to_datetime(
    Monthly_Plan["Month"],
    format="%b-%Y"
)

In [17]:
print(Monthly_Plan["Month"].isnull().sum())

0


In [18]:
print(Monthly_Plan["Month"].min())
print(Monthly_Plan["Month"].max())

2024-04-01 00:00:00
2025-03-01 00:00:00


In [22]:
Monthly_Plan["Month"]=Monthly_Plan["Month"].dt.strftime("%b-%Y")

In [23]:
Monthly_Plan.head()

,Month,Site ID,Product Category,Forecasted Sales,Actual Sales,Seasonal Adjustments
0,May-2024,PS3Y,Dairy,81260.50,96315.510888,0.1853
1,Dec-2024,W1Q5,Apparel,60609.93,64357.950697,0.0618
2,Sep-2024,T4UF,Apparel,84277.99,94950.573265,0.1266
3,Dec-2024,T4UF,Apparel,79067.48,82228.634271,0.0400
4,Mar-2025,OTOH,Apparel,50727.92,42559.446989,-0.1610


In [24]:
print(Monthly_Plan["Product Category"].value_counts())
print(Monthly_Plan["Product Category"].isnull().sum())

Product Category
Apparel        37
Bakery         31
Dairy          26
Electronics    26
Name: count, dtype: int64
0


In [25]:
Monthly_Plan["Product Category"] = (
    Monthly_Plan["Product Category"].str.strip()
)

In [27]:
print((Monthly_Plan[["Forecasted Sales","Actual Sales","Seasonal Adjustments"]] <= 0).sum())

Forecasted Sales         0
Actual Sales             0
Seasonal Adjustments    63
dtype: int64


In [28]:
print(Monthly_Plan["Seasonal Adjustments"].describe())

count    120.000000
mean      -0.007223
std        0.111414
min       -0.199500
25%       -0.100500
50%       -0.010250
75%        0.089575
max        0.198700
Name: Seasonal Adjustments, dtype: float64


In [29]:
print(
    Monthly_Plan[
        (Monthly_Plan["Seasonal Adjustments"] < -1) |
        (Monthly_Plan["Seasonal Adjustments"] > 1)
    ]
)

Empty DataFrame
Columns: [Month, Site ID, Product Category, Forecasted Sales, Actual Sales, Seasonal Adjustments]
Index: []


In [31]:
print(Monthly_Plan.duplicated().sum())
print(Monthly_Plan.duplicated().sum())

0
0


In [32]:
Monthly_Plan[
    Monthly_Plan.duplicated(
        subset=["Month", "Site ID", "Product Category"],
        keep=False
    )
]

,Month,Site ID,Product Category,Forecasted Sales,Actual Sales,Seasonal Adjustments
50,Feb-2025,BVM2,Electronics,83507.89,69215.364087,-0.1712
74,Nov-2024,AM14,Apparel,51656.35,48737.592703,-0.0565
80,Feb-2025,BVM2,Electronics,79767.75,75486.411613,-0.0537
105,Nov-2024,AM14,Apparel,63688.00,52704.186571,-0.1725


In [33]:
Monthly_Plan[
    Monthly_Plan.duplicated(keep=False)
]

,Month,Site ID,Product Category,Forecasted Sales,Actual Sales,Seasonal Adjustments


In [34]:
Monthly_Plan.duplicated(
    subset=["Month", "Site ID", "Product Category"]
).sum()

np.int64(2)

In [35]:
Monthly_Plan.loc[
    Monthly_Plan.duplicated(
        subset=["Month", "Site ID", "Product Category"],
        keep=False
    ),
    ["Month", "Site ID", "Product Category"]
]

,Month,Site ID,Product Category
50,Feb-2025,BVM2,Electronics
74,Nov-2024,AM14,Apparel
80,Feb-2025,BVM2,Electronics
105,Nov-2024,AM14,Apparel


In [36]:
valid_categories = [
    "Apparel",
    "Electronics",
    "Footwear",
    "Dairy"
]

print(
    Monthly_Plan[
        ~Monthly_Plan["Product Category"].isin(valid_categories)
    ]
)

        Month Site ID Product Category  Forecasted Sales  Actual Sales  \
5    Apr-2024    7EEQ           Bakery          40722.14  46095.571110   
7    Mar-2025    0L72           Bakery          76825.98  74035.990980   
8    Jul-2024    4CIY           Bakery          20867.12  23806.832888   
14   Mar-2025    KJLT           Bakery          83371.97  70150.975096   
16   Aug-2024    2VLP           Bakery          45808.30  42928.951618   
19   Jul-2024    NY20           Bakery          28874.14  31350.207638   
32   Feb-2025    A1WJ           Bakery          73416.62  77752.088300   
33   Mar-2025    XDP5           Bakery          96786.08  88817.787407   
36   Jan-2025    0T0P           Bakery          86141.28  69107.535697   
37   Apr-2024    T4UF           Bakery          23833.30  22142.375317   
40   Oct-2024    CX19           Bakery          58877.21  56602.596535   
41   Jul-2024    8CGV           Bakery          69700.10  64990.228283   
42   Jan-2025    M5IG           Bakery

In [37]:
Monthly_Plan["Product Category"].value_counts()

Product Category
Apparel        37
Bakery         31
Dairy          26
Electronics    26
Name: count, dtype: int64

In [38]:
Monthly_Plan["Difference"] = (
    Monthly_Plan["Actual Sales"]
    - Monthly_Plan["Forecasted Sales"]
)

print(Monthly_Plan["Difference"].describe())

count      120.000000
mean      -487.642508
std       6897.165905
min     -17033.744303
25%      -4761.057930
50%       -706.982218
75%       3225.299558
max      15055.010888
Name: Difference, dtype: float64


In [39]:
subset=["Month","Site ID","Product Category"]

In [40]:
subset=[
    "Month",
    "Site ID",
    "Product Category",
    "Forecasted Sales",
    "Actual Sales"
]

In [41]:
Monthly_Plan.columns = Monthly_Plan.columns.str.strip()

print(Monthly_Plan.columns.tolist())

['Month', 'Site ID', 'Product Category', 'Forecasted Sales', 'Actual Sales', 'Seasonal Adjustments', 'Difference']


In [42]:
Monthly_Plan.to_csv(
    "../03_Cleaned_Data/Monthly_Seasonal_Planning_Cleaned.csv",
    index=False
)

## Monthly Seasonal Planning Data Validation

- Total Records: 120
- No Missing Values
- No Duplicate Rows
- Month Converted to Datetime
- Product Categories Validated
- Forecasted Sales Validated
- Actual Sales Validated
- Seasonal Adjustments Validated
- Column Names Cleaned
- Dataset Ready for SQL & Power BI